api/controller.go

```diff
@@ -1265,6 +1265,56 @@ func writeSymlinkToS3(params metadata.CreateSymlinkParams, repo *catalog.Reposit
 	return err
 }
 
+func (c *Controller) MetadataGetMetarangeHandler() metadata.GetMetaRangeHandler {
+	return metadata.GetMetaRangeHandlerFunc(func(params metadata.GetMetaRangeParams, user *models.User) middleware.Responder {
+		deps, err := c.setupRequest(user, params.HTTPRequest, []permissions.Permission{
+			{
+				// TODO(oz): May want tighter permissions here.
```

Comment:
Remember to remove the todo after verifying

Reply:
After f2fing, changed to require _both_ list objects _and_ read repo: it gives a cross between the 2.  E.g. with (just) read permission on underlying storage it gives list objects (and without it gives size of branch), which is list-objects-like.  But it _also_ gives the storage namespace for the repo, which is read-repo-like.  So best to require both.